crates/primitives/src/basefee.rs

```diff
@@ -0,0 +1,16 @@
+use alloy_eips::eip1559::BaseFeeParams;
+use reth_primitives::basefee::calc_next_block_base_fee;
+
+use crate::MIN_BASE_FEE_PER_GAS;
+
+pub fn calculate_next_block_base_fee(
+    gas_used: u128,
+    gas_limit: u128,
+    base_fee: Option<u64>,
+    base_fee_params: BaseFeeParams,
+) -> Option<u64> {
+    Some(std::cmp::max(
```

Comment:
why option?

+ shouldn't this return u128?

Reply:
The old way we calc basefee is usually something like this:
```rust
parent_block
.header
.next_block_base_fee(cfg.base_fee_params)
```
the function  next_block_base_fee returns `Option<u64>` that's why I tried to make it sth like that

also  pretty much everywhere basefee is u64

Reply:
ok, I'll respect that